packages/techdocs-cli/src/commands/generate/generate.ts

```diff
@@ -79,9 +84,8 @@ export default async function generate(cmd: Command) {
   await techdocsGenerator.run({
     inputDir: sourceDir,
     outputDir,
-    ...(cmd.techdocsRef && {
-      parsedLocationAnnotation
-    })
+    ...(cmd.techdocsRef ? parsedLocationAnnotation : {}),
```

Comment:
This ensures the additional logger argument is type checked. The previous syntax treats the object as any and bypasses all type checking.